clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java

```diff
@@ -716,11 +719,14 @@ private void handleShareFetchSuccess(Node fetchTarget,
             final Map<TopicIdPartition, ShareFetchResponseData.PartitionData> responseData = new LinkedHashMap<>();
 
             response.data().responses().forEach(topicResponse ->
-                    topicResponse.partitions().forEach(partition ->
-                            responseData.put(new TopicIdPartition(topicResponse.topicId(),
-                                    partition.partitionIndex(),
-                                    metadata.topicNames().getOrDefault(topicResponse.topicId(),
-                                            topicNamesMap.remove(new IdAndPartition(topicResponse.topicId(), partition.partitionIndex())))), partition))
+                    topicResponse.partitions().forEach(partition -> {
+                        TopicIdPartition tip = createTopicIdPartition(topicResponse.topicId(), partition.partitionIndex());
+                        if (tip != null) {
+                            responseData.put(tip, partition);
+                        } else {
+                            log.error("Topic name not found for topic ID {}", topicResponse.topicId());
```

Comment:
In this case, you'll log two error lines because `createTopicIdPartition` write a log line. I suggest choosing whether you want to log in the caller or in the method and using the same pattern for all instances.